mne/gui/_ieeg_locate_gui.py

```diff
@@ -1034,3 +1034,8 @@ def _update_moved(self):
             *self._ras))
         self._intensity_label.setText('intensity = {:.2f}'.format(
             self._ct_data[tuple(self._current_slice)]))
+
+    def closeEvent(self, event):
+        """Clean up upon closing the window."""
+        self._renderer.plotter.close()
+        self.close()
```

Comment:
One thing you have to be careful about in any overridden PyQt method is that an error in here will cause a segfault (as you saw in the tests). It's thus a good idea to enclose these in a try/except, and also be more careful about whether some of these can be `None` (for example if they get their own closeEvent sooner and close themselves, could `_renderer.plotter` be `None` by this point?).

We can tackle that later, though -- for example, we maybe want a `safe_event` decorator or overridden methods that make sure they *never* emit a RuntimeError, because we never want to cause a seg fault for users (a hung window that can't be used is already better than a segfault, which kills the terminal).

Reply:
... looks like we already have `safe_event` in `_brain.py` that we should probably just move to `utils` so you can make use of it here

Reply:
I'll change to using `safe_event` actually